chore(memory): promote strong-fit configuration drafts from memory-pipeline for review (#130)
medic/cht-agent: agent-memory/domains/configuration/issues/10198-fix8026-validate-empty-branding-doc.md

---
id: cht-core-8026
category: bug
domain: configuration
domainFit: strong
issueNumber: 8026
issueUrl: https://github.com/medic/cht-core/issues/8026
title: Validate empty branding doc to prevent exception in admin branding controller

> lastUpdated: '2026-06-22'

Reviewer: nitpick: 10198, 9696 and 9727 keep `lastUpdated: '2026-06-22'` despite substantive edits in this iteration, while the five other configuration drafts touched in the same commits were bumped to '2026-07-28'.

## Code Patterns

Guard branding-doc property access with existence checks and initialize a minimal default object shape before binding to $scope (admin/src/js/controllers/images-branding.js), so the template (admin/src/templates/images_branding.html) renders safely against empty/partial docs.

## Design Choices

Rather than failing on incomplete branding docs, the controller defensively scaffolds the minimum keys the template requires — tolerating empty/partial input from external tooling (cht-conf) instead of assuming a fully-populated document.

## Related Files

- .gitignore
- admin/src/js/controllers/images-branding.js
- admin/src/templates/images_branding.html
- admin/tests/unit/controllers/images-branding.spec.js

## Testing

> Added a new unit test file (admin/tests/unit/controllers/images-branding.spec.js) for the previously untested controller, following conventions of existing controller tests, exercising the empty-branding-doc scenario. One related case is not covered by the fix.

Reviewer: issue: stale mid-review artifact. The merged spec covers both cases: 'should not fail when branding doc is empty' and 'should not fail when resources is null' (the case flagged in review got an ng-if plus that second test before merge). Nothing known is left uncovered.

```suggestion
Added a new unit test file (admin/tests/unit/controllers/images-branding.spec.js) for the previously untested controller, following conventions of existing controller tests, exercising both the empty-branding-doc and null-resources scenarios.
```
